Azure/GCP

benmoss opened the pull request:
**What this PR does / why we need it**:
Adds support for Azure/GCP Windows cloud providers

**How can this PR be verified?**
Deploy Windows workers to those environments

**Is there any change in kubo-release?**
No

**Is there any change in kubo-ci?**
No

**Does this affect upgrade, or is there any migration required?**
No

**Which issue(s) this PR fixes:**
None

**Release note**:
```release-note
Add opsfiles for configuring Windows workers on Azure and GCP
```

This is built on top of #392

astrieanna committed:
Add opsfile for setting windows overlay vni/port

Signed-off-by: Ben Moss <bmoss@pivotal.io>

benmoss committed:
Add opsfiles for GCP windows support

benmoss committed:
Add Windows/Azure cloud provider opsfile

zankich commented:
A story has been created for this in our tracker https://www.pivotaltracker.com/story/show/165840796

neil-hickey commented on manifests/ops-files/iaas/gcp/windows/ephemeral-disk.yml:
```diff
@@ -0,0 +1,5 @@
+---
+- type: replace
+  path: /instance_groups/name=windows-worker/vm_extensions?/-
+  value: ((windows_ephemeral_disk_type))
```
if disk_size too small --> refuses to make it a windows vm (maybe we will need this)

neil-hickey commented on manifests/ops-files/windows/increase-disk.yml:
```diff
@@ -0,0 +1,5 @@
+---
+- type: replace
+  path: /instance_groups/name=windows-worker/vm_extensions?/-
+  value: ((deployment_name))-windows-worker-cloud-properties
```
we don't need this

benmoss committed:
Add aws cloud provider for Windows nodes

rumshenoy commented on manifests/ops-files/windows/use-overlay.yml:
```diff
@@ -0,0 +1,7 @@
+  value: 4096
+- type: replace
+  path: /instance_groups/name=worker/jobs/name=flanneld/properties?/port
+  value: 4789
```
Should these values also be applied to the master instance_group? What about the windows-worker group?

rumshenoy replied:
We applied these properties to the masters and that fixed our metrics server issue.

rumshenoy replied:
It might also make sense to rename this ops-file. We normally use overlay, this is just changing the flanneld properties that windows needs.

benmoss replied:
Updated it to be called `overlay-overrides.yml` and included the master instance group. We didn't bother adding `etcdctl` to the Windows job, so it assumes that the configuration has already been written to etcd by one of the Linux VMs. This is a pretty safe guarantee, since the masters run flannel and will come up before the Windows node.

benmoss committed:
Allow prioritizing InternalIP over hostname for kubelet communication

Required for Windows nodes on Azure, where the VM name in Azure doesn't
resolve via DNS.

https://github.com/kubernetes/kubernetes/issues/79232
https://github.com/kubernetes/kubernetes/issues/22770#issuecomment-266215007

benmoss committed:
Remove privileged flag from windows kubelet

Doesn't really do anything since privileged containers aren't supported
anyway
